repository/src/lib.rs

```diff
@@ -196,8 +196,88 @@ impl<T: RawRepository> DistributedRepository<T> {
     ///
     /// Note that the proof in the `fp` branch must be set for the candidate commit
     /// for the last finalized block, which is `block_commit`.
-    pub async fn sync(&mut self, _block_commit: &CommitHash) -> Result<(), Error> {
-        unimplemented!()
+    pub async fn sync(&mut self, block_commit: &CommitHash) -> Result<(), Error> {
+        // See if this commit is a descendant of the current finalized branch
+        let current_finalized_commit = self.raw.locate_branch(FINALIZED_BRANCH_NAME.into()).await?;
+        if self
+            .raw
+            .find_merge_base(current_finalized_commit, *block_commit)
+            .await?
+            != current_finalized_commit
+        {
+            return Err(anyhow!(
+                "block commit is not a descendant of the current finalized branch"
+            ));
+        }
+
+        // Construct a commit list starting from the next commit of the last finalized block to the `branch_tip_commit`(the most recent commit of the branch)
```

Comment:
Trim this long comment please

refer to https://github.com/postech-dao/simperby/pull/197/commits/f65178195edfca265f3db0ab2ae613fc746b7df9

Reply:
Also please update this process using query_commit_path.

refer to https://github.com/postech-dao/simperby/pull/197

Reply:
fixed.